sirepo/package_data/static/js/controls.js

```diff
@@ -419,14 +427,29 @@ SIREPO.app.controller('ControlsController', function(appState, controlsService,
     });
     $scope.$on('controlSettings.changed', () => {
         for (const m in appState.models) {
-            if (m.includes('instrumentAnimation')) {
+            if (m.includes('instrumentAnimation') && m != 'instrumentAnimationAll') {
                 appState.models[m].particlePlotSize = appState.models.controlSettings.particlePlotSize;
                 appState.saveQuietly(m);
             }
         }
     });
     $scope.$on('initialMonitorPositionsReport.changed', getInitialMonitorPositions);
-
+    $scope.$on('instrumentAnimationAll.changed', () => {
+        if (!self.instrumentAnimations){
+            return;
+        }
+        self.instrumentAnimations.forEach((e, i) => {
+            if (e.modelKey == 'instrumentAnimationAll' || e.modelKey == 'instrumentAnimationTwiss'){
+                return;
+            }
+            for (const key in appState.models[e.modelKey]) {
+                if (key != 'id'){
+                    appState.models[e.modelKey][key] = appState.models.instrumentAnimationAll[key];
+                }
+            }
+            appState.saveChanges(e.modelKey);
```

Comment:
Here I would collect up all the model names that changed in an array m and then call saveChanges(m). Each call to saveChanges() does a lot of work (and maybe a server call), so in the forEach loop, add m.push(e.modelKey) and after the loop add if (m.length) { appState.saveChanges(m) }